pages/docs/guides/singleton.mdx

```diff
@@ -0,0 +1,94 @@
+import { VersionBadge, Callout, ImageTheme } from "src/shared/Docs/mdx";
+
+# Singleton Functions <VersionBadge version="TypeScript v3.38.0+" />
+
+Singleton Functions enable you to ensure that only a single run of your function (_or a set of specific function runs, based on specific event properties_) is happening at a time.
+
+<ImageTheme
+  dark={'/assets/docs/features/inngest-functions/flow-control/singleton-functions/diagram.png'}
+  light={'/assets/docs/features/inngest-functions/flow-control/singleton-functions/diagram.png'}
+  className="my-0"
+  alt={'Singleton Functions only process one run at a time.'}
+/>
+
+<Callout variant="info">
+
+    Singleton Functions are available in the TypeScript SDK starting from version 3.38.0.
+
+</Callout>
+
+## When to use Singleton Functions
+
+Singleton Functions are useful when you want to ensure that only a single instance of a function is running at a time, for example:
+- A third-party data synchronization workflow
+- A compute- or time-intensive function that should not be run multiple times at the same time (ex: AI processing)
+
+
+### Singleton compared to concurrency:
+
+While [Concurrency](/docs/guides/concurrency) set to `1` ensures that only a single step of a given function is running at a time, Singleton Functions ensure that only a single run of a given function is happening at a time.
+
+
+### Singleton compared to Rate Limiting:
+
+[Rate Limiting](/docs/guides/rate-limiting) is similar to Singleton Functions, but it is designed to limit the number of concurrent runs of a function, whereas Singleton Functions are designed to ensure that only a single run of a function occurs over a given time window.
+
+Rate Limiting is useful for controlling the rate of execution of a function, while Singleton Functions are useful for ensuring that only a single run of a function occurs over a given time window.
+
+
+
+
+
+
+## How it works
+
+Singleton Functions are configured using the `singleton` property in the function definition.
+
+The following `data-sync` function will skip new runs if another run of the same function is already running for the same user:
```

Comment:
suggestion - this is the first place that we reference "key" support, I think we should call that out explicitly either above or below this, similar to something like this: https://www.inngest.com/docs/guides/debounce#using-a-key